src/modules/finance/hooks/useAssetListData/useAssetListData.ts

```diff
@@ -8,7 +8,33 @@ export const useAssetListData = (params: IGetAssetListParams) => {
 
     const { data: assetListData, status, fetchStatus, isFetchingNextPage, fetchNextPage } = useAssetList(params);
 
-    const assetList = assetListData?.pages.flatMap((page) => page.data);
+    const assetList = assetListData?.pages
+        .flatMap((page) => page.data)
+        .map((asset) => {
+            const amount = asset.amount ?? '0';
+            const hasAmount = asset.amount != null && Number(amount) > 0;
+            const hasAmountUsd = asset.amountUsd != null && Number(asset.amountUsd) > 0;
+            const price = asset.token.priceUsd;
+            let derivedPrice = price;
+
+            if ((!price || Number(price) === 0) && hasAmountUsd && hasAmount) {
+                const perToken = Number(asset.amountUsd) / Number(amount);
+                if (perToken > 0) {
+                    derivedPrice = perToken.toString();
+                }
+            }
+
+            return {
+                ...asset,
+                amount,
+                token: {
+                    ...asset.token,
+                    name: asset.token.name || 'Unknown',
+                    symbol: asset.token.symbol || 'UNKNOWN',
+                    priceUsd: derivedPrice || asset.token.priceUsd || '0',
+                },
+            };
+        });
```

Comment:
optional idea: replacing `flatMap/map` with simple nested `for` loops keeps the same overall complexity (still `O(totalAssets)`) but avoids creating intermediate arrays and running many callback functions

it speeds up asset processing a bit (cuts a bunch of redundant allocations and GC work), but this isn’t a required change (especially on our datasets size)

feel free to skip if you prefer the current style

```suggestion
const assetList = [];
if (assetListData?.pages) {
  for (const page of assetListData.pages) {
    for (const asset of page.data) {

      const amount = Number(asset.amount) || 0;
      const amountUsd = Number(asset.amountUsd) || 0;
      const price = Number(asset.token.priceUsd) || 0;

      let finalPrice = price;
      if (finalPrice === 0 && amount > 0 && amountUsd > 0) {
        finalPrice = amountUsd / amount;
      }

      assetList.push({
        ...asset,
        amount: String(amount),
        token: {
          ...asset.token,
          name: asset.token.name || "Unknown",
          symbol: asset.token.symbol || "UNKNOWN",
          priceUsd: String(finalPrice),
        },
      });
    }
  }
}
```